DiffusionPreprocessing/DiffPreprocPipeline_Split.sh

```diff
@@ -0,0 +1,385 @@
+				index=$(( index + 1 ))
+				;;
+			--dwiname=*)
+				DWIName=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			*)
+				PassOnArguments+=" ${argument} "
+				index=$(( index + 1 ))
+				;;
+		esac
+	done
+	
+	local error_msgs=""
+
+	# check required parameters
+	if [ -z ${PosInputImages} ] ; then
+		error_msgs+="\nERROR: <positive-phase-encoded-data> not specified"
+	fi
+	
+	if [ -z ${NegInputImages} ] ; then
+		error_msgs+="\nERROR: <negative-phase-encoded-data> not specified"
+	fi
+
+	if [ -z ${StudyFolder} ] ; then
+		error_msgs+="\nERROR: <study-path> not specified"
+	fi
+
+	if [ -z ${Subject} ] ; then
+		error_msgs+="\nERROR: <subject-id> not specified"
+	fi
+
+	if [ ! -z "${error_msgs}" ] ; then
+		usage
+		echo -e ${error_msgs}
+		echo ""
+		exit 1
+	fi
+	
+	# report parameters
+	echo "-- ${SCRIPT_NAME}: Specified Command-Line Parameters - Start --"
+	echo "   PosInputImages: ${PosInputImages}"
+	echo "   NegInputImages: ${NegInputImages}"
+	echo "   DWIName: ${DWIName}"
+	echo "   Arguments for DiffPreprocPipeline.sh: ${PassOnArguments}"
+	echo "-- ${SCRIPT_NAME}: Specified Command-Line Parameters - End --"
+}
+
+# 
+# Function Description
+#  Validate necessary environment variables
+#
+validate_environment_vars()
+{
+	local error_msgs=""
+
+	# validate
+	if [ -z ${HCPPIPEDIR_dMRI} ] ; then
+		error_msgs+="\nERROR: HCPPIPEDIR_dMRI environment variable not set"
+	fi
+
+	if [ ! -e ${HCPPIPEDIR}/DiffusionPreprocessing/DiffPreprocPipeline.sh ] ; then
+		error_msgs+="\nERROR: HCPPIPEDIR/DiffusionPreprocessing/DiffPreprocPipeline.sh not found"
+	fi
+
+	if [ -z ${FSLDIR} ] ; then
+		error_msgs+="\nERROR: FSLDIR environment variable not set"
+	fi
+	
+	if [ ! -z "${error_msgs}" ] ; then
+		usage
+		echo -e ${error_msgs}
+		echo ""
+		exit 1
+	fi
+
+	# report
+	echo "-- ${SCRIPT_NAME}: Environment Variables Used - Start --"
+	echo "   HCPPIPEDIR_dMRI: ${HCPPIPEDIR_dMRI}"
+	echo "   FSLDIR: ${FSLDIR}"
+	echo "-- ${SCRIPT_NAME}: Environment Variables Used - End --"
+}
+
+#
+# Function description
+#  Main processing of script if --split is set
+#
+main()
+{
+	# Get Command Line Options
+	get_options $@
+
+	# Validate environment variables
+	validate_environment_vars $@
+
+	# Establish tool name for logging
+	log_SetToolName "${SCRIPT_NAME}"
+
+
+	PosInputImages=`echo ${PosInputImages} | sed 's/@/ /g'`
+	log_Msg "PosInputImages: ${PosInputImages}"
+	Pos_count=`echo ${PosInputImages} | wc -w`
+	echo ${Pos_count}
+
+	NegInputImages=`echo ${NegInputImages} | sed 's/@/ /g'`
+	log_Msg "NegInputImages: ${NegInputImages}"
+	Neg_count=`echo ${NegInputImages} | wc -w`
+	echo ${Neg_count}
+
+	# verify positive and negative datasets are provided in pairs
+	if [ ${Pos_count} -ne ${Neg_count} ] ; then
+		log_Msg "Wrong number of input datasets! Make sure that you provide pairs of input filenames."
+		exit 1
+	fi
+
+	# verify that none of the files are empty
+	for Image in ${PosInputImages} ; do
+		if [ ${Image} = EMPTY ] ; then
```

Comment:
Does `EMPTY` have a predefined meaning in bash?  I couldn't find anything on this construction on the web...